test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -0,0 +1,241 @@
+		__dirname,
+		'..',
+		'..',
+		'..',
+		'assets',
+		'10x10_e2e_test_image_z9T8jK.png'
+	);
+	const filename = uuid();
+	const tmpFilename = path.join( os.tmpdir(), filename + '.png' );
+	fs.copyFileSync( testImagePath, tmpFilename );
+	return tmpFilename;
+}
+
+async function getBackgroundColorAndOpacity( locator ) {
+	return await locator.evaluate( ( el ) => {
+		const computedStyle = window.getComputedStyle( el );
+		return [ computedStyle.backgroundColor, computedStyle.opacity ];
+	} );
+}
+
+test.describe( 'Cover', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'can set overlay color using color picker on block placeholder', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Locate the Black color swatch.
+		const blackColorSwatch = page.getByRole( 'button', {
+			name: 'Color: Black',
+		} );
+		await expect( blackColorSwatch ).toBeVisible();
+
+		// Get the RGB value of Black.
+		const blackRGB = await blackColorSwatch.evaluate(
+			( node ) => node.style.backgroundColor
+		);
+
+		// Create the block by clicking selected color button.
+		await blackColorSwatch.click();
+
+		// Get the RGB value of the background dim.
+		// The span element implements the colored background.
+		const backgrounDimLocator = page
+			.getByRole( 'document', { name: 'Block: Cover' } )
+			.locator( 'span[aria-hidden="true"]' );
+		const actualRGB = await backgrounDimLocator.evaluate(
+			( node ) => node.style.backgroundColor
+		);
+
+		expect( blackRGB ).toEqual( actualRGB );
+	} );
+
+	test( 'can set background image using image upload on block placeholder', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		const testImage = await getTestImage();
+		const testImageFilename = path.basename( testImage );
+
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		await coverBlock
+			.getByTestId( 'form-file-upload-input' )
+			.setInputFiles( testImage );
+
+		// Wait for the img's src attribute to be prefixed with http.
+		// Otherwise, the URL for the img src attribute is that of a placeholder.
+		await coverBlock.locator( `img[src^=http]` ).waitFor();
+
+		const backgroundImageURL = await coverBlock
+			.locator( 'img' )
+			.getAttribute( 'src' );
+
+		expect( backgroundImageURL ).toContain( testImageFilename );
+	} );
+
+	test( 'dims background image down by 50% by default', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		const testImage = await getTestImage();
+
+		const coverBlock = page.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		await coverBlock
+			.getByTestId( 'form-file-upload-input' )
+			.setInputFiles( testImage );
+
+		const [ backgroundDimColor, backgroundDimOpacity ] =
+			await getBackgroundColorAndOpacity(
+				coverBlock.locator( 'span[aria-hidden="true"]' )
```

Comment:
Hmm, this is a bit awkward, as we're trying to test something that really should be tested by visual testing instead. I guess adding a comment about why we have to use CSS selector above would help the readers?

Reply:
@kevin940726 Do you think this particular test case is not appropriate in the setting of E2E? If so, I'm happy to remove it from the migrated spec.

Side note, computing the style for the Cover Block locator returns an opacity value of 1, while computing the style for the hidden `span` that actually implements the background returns an opacity of 0.5, resulting in a discrepancy.

I am happy to write comments explaining why a CSS span has to be used instead of the cover block. How should I proceed?

Reply:
I think adding a comment above will be better for now. In case this was testing for an actual regression and removing it prematurely put us at risk of introducing the bug again. 👍

Reply:
Another idea would be to capture the screenshot for this element and do raw pixel manipulation to find the exact color code. This is a lot of work though, and I'm okay to just skip it and do a simpler assertion for this tiny step 😆 . At least until we make this easy to test 😛 .

Reply:
> do raw pixel manipulation to find the exact color code. This is a lot of work though,

I agree that this is too much work for a test step, especially when considering that checking for a hidden span would be much simpler.

> I think adding a comment above will be better for now

I've added comments in both places where the `span[aria-hidden="true"]` selector is used, explaining the reason.

Fixed in d6a4f4499555c768f60a9ca86ae70cc0746e1332.